pkg/cmd/server/api/helpers.go

```diff
@@ -17,6 +17,71 @@ import (
 	cmdutil "github.com/openshift/origin/pkg/cmd/util"
 )
 
+var (
+	knownOpenShiftFeatureSet map[string]string
+)
+
+func init() {
+	knownOpenShiftFeatureSet = make(map[string]string, len(KnownOpenShiftFeatures))
+	for _, feature := range KnownOpenShiftFeatures {
+		knownOpenShiftFeatureSet[strings.ToLower(feature)] = feature
+	}
+}
+
+// Add extends feature list with given valid items. They are appended
+// unless already present.
+func (fl *FeatureList) Add(items ...string) error {
+	unknown := []string{}
+	toAppend := make([]string, 0, len(items))
+	for _, item := range items {
+		feature, exists := knownOpenShiftFeatureSet[strings.ToLower(item)]
+		if !exists {
+			unknown = append(unknown, item)
+			continue
+		}
+		if fl.Has(feature) {
+			continue
+		}
+		toAppend = append(toAppend, feature)
+	}
+	if len(unknown) > 0 {
+		return fmt.Errorf("unknown OpenShift features: %s", strings.Join(unknown, ", "))
```

Comment:
just "unknown features: %s".  We're trying to avoid explicit mention of openshift.